src/CRISP.BackendChallenge/Repository/ContextRepository.cs

```diff
@@ -18,38 +19,69 @@ public IQueryable<T> Query()
     }
 
     /// <inheritdoc />
-    public IEnumerable<T> GetAll()
+    public IEnumerable<Employee> GetAll()
```

Comment:
You have effectively committed this class to _not_ being generic by making explicit references to a certain concrete class (`Employee`). This itself is not bad, but you have left the class and a few of its other methods generic. It doesn't make sense for part of the class to be generic, and the other part to require a specific entity type.

Reply:
Changed back to generic so it falls in line with the rest of the IRepository